stripe/src/main/java/com/stripe/android/EphemeralKeyManager.java

```diff
@@ -0,0 +1,108 @@
+package com.stripe.android;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import java.lang.ref.WeakReference;
+import java.util.Calendar;
+import java.util.concurrent.TimeUnit;
+
+class EphemeralKeyManager {
+
+    private @Nullable EphemeralKey mEphemeralKey;
+    private @NonNull EphemeralKeyProvider mEphemeralKeyProvider;
+    private @Nullable Calendar mOverrideCalendar;
+    private @NonNull KeyManagerListener mListener;
+    private final long mTimeBufferInSeconds;
+
+    EphemeralKeyManager(
+            @NonNull EphemeralKeyProvider ephemeralKeyProvider,
+            @NonNull KeyManagerListener keyManagerListener,
+            long timeBufferInSeconds,
+            @Nullable Calendar overrideCalendar) {
+
+        mEphemeralKeyProvider = ephemeralKeyProvider;
+        mListener = keyManagerListener;
+        mTimeBufferInSeconds = timeBufferInSeconds;
+        mOverrideCalendar = overrideCalendar;
+        retrieveEphemeralKey();
+    }
+
+    void retrieveEphemeralKey() {
+        if (mEphemeralKey != null &&
```

Comment:
Ahh. Yeah, I think I used to try and access the time and only pass that to the expiry check function.